### PR TITLE
chore(deps): upgrade packages

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

upgrade packages to prepare the React 18 update.

- upgrade configs to 1.0.0 rc
- update code to fit the new rule set accordingly

## Testing

CI + locally. console sign-in/out works.

### CODE REVIEW

File: packages/console/package.json [hunk 1]
```diff
@@ -90,20 +90,28 @@
...
       "@typescript-eslint/prefer-nullish-coalescing": "off"
-    }
+    },
+    "overrides": [
+      {
+        "files": [
+          "*.test.ts"
+        ],
+        "rules": {
+          "@typescript-eslint/no-unsafe-return": "off"
```
Comment (demonzoo) [inline packages/console/package.json:100]: Do we have unsafe returns in our ac code?

Comment (gao-sun) [reply packages/console/package.json:100]: a lot in test files

Comment (demonzoo) [reply packages/console/package.json:100]: Hmmm, but we don't have test files in console, do we?

File: packages/console/package.json [hunk 2]
```diff
@@ -90,20 +90,28 @@
...
+        }
+      },
+      {
+        "files": [
+          "*.d.ts"
+        ],
+        "rules": {
+          "@typescript-eslint/consistent-type-definitions": "off"
```
Comment (demonzoo) [inline packages/console/package.json:108]: Should this be turned off in ts-config? Seems like we don't want to limit the usages of `type` and `interface`, though we do prefer `type` over `interface` in practice

File: packages/console/package.json (hunk 1 above)
Comment (gao-sun) [reply packages/console/package.json:100]: ah, this is a mistake

File: packages/console/package.json (hunk 2 above)
Comment (gao-sun) [reply packages/console/package.json:108]: personally i'd like to have this

Comment (demonzoo) [reply packages/console/package.json:108]: But I just checked our eslint-config and seems it's already in there?

image

**Review by charIeszhao — APPROVED** [review]
Overall looks good. Please check my comments before merging

**Review by darcyYe — APPROVED** [review]

**Review by wangsijie — APPROVED** [review]

File: packages/console/package.json (hunk 2 above)
Comment (gao-sun) [reply packages/console/package.json:108]: yes, it's another mistake. should be for testing only. will remove

**Pull request merged**